### ISSUE TITLE
Object detection with YOLOv8n

### ISSUE THREAD

**Issue by Hewgo2545** [ultralytics/ultralytics#21847]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I’m a beginner in machine learning and AI. Please suggest tips and know-how to improve my skills.

Is it feasible to train for 50 epochs on an object-detection task (car body type: sedan, SUV, hatchback) that focuses on the rear of the car and achieve over 90% accuracy? I plan to run the system on an NVIDIA Jetson Nano with a Logitech webcam.

My questions:

1.How can I handle noisy backgrounds, changing ambient light (daytime vs. evening), and car color? I don’t want the model to rely on color—only on shape.

2.How should I prepare the dataset—for example, converting images to grayscale before training and applying augmentation?

3.I trained a demo model for 50 epochs, and I think it is overfitting.

result here:

<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/da8001fc-20a3-4b75-a8b1-e3dd68efa02b" />

F1 Score:

<img width="2250" height="1500" alt="Image" src="https://github.com/user-attachments/assets/3bd6028b-7ef2-4e03-84f4-16d5e00b2272" />

Precision and Confidence curve :

<img width="2250" height="1500" alt="Image" src="https://github.com/user-attachments/assets/95ecf6dc-8324-45ca-b279-78229e95d1db" />

Precision and recall :

<img width="2250" height="1500" alt="Image" src="https://github.com/user-attachments/assets/b3a68079-b221-4311-857f-9dc66cc651a1" />

Recall and Confidence Curve:

<img width="2250" height="1500" alt="Image" src="https://github.com/user-attachments/assets/e3370672-746c-4804-b461-cbc1a8e30afa" />

Please help me. 
Appreciate

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Hewgo2545, thank you for your interest in Ultralytics 🚀! This is an automated reply to help you get the fastest possible support. An Ultralytics engineer will also review and assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us better understand your training question and the overfitting you suspect, please share:
-Exact command(s) used to train/val/predict
-ultralytics and PyTorch versions, OS, CUDA/CUDNN version, and device (e.g., Jetson Nano details)
-Model used (e.g., yolov8n), image size, epochs, batch size, optimizer, and key hyperparameters
-data.yaml, number of images per split, and class distribution
-Examples of training/validation images (especially showing rear-only views, lighting conditions, and backgrounds)
-Training logs and full results (metrics, confusion matrix, PR/F1 curves)
-Any augmentations applied and preprocessing steps performed (e.g., grayscale conversion)
-Deployment target details (intended inference size/FPS on Jetson Nano, webcam input resolution)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
> Is it feasible to train for 50 epochs on an object-detection task and take around 10 pics for 1 model to training (car body type: sedan, SUV, hatchback) that focuses on the rear of the car and achieve over 90% accuracy?

The model would very likely perform poorly with that few images. You need hundreds of images for each class, if not thousands.

> 1.How can I handle noisy backgrounds, changing ambient light (daytime vs. evening), and car color? I don’t want the model to rely on color—only on shape. There are 4 colors as white, black, red, and grey.

If you have enough data with those variations, then this wouldn't be an issue.

> 2.How should I prepare the dataset—for example, converting images to grayscale before training and applying augmentation?

No. Don't apply augmentations before training. Ultralytics performs augmentations automatically.

> 3.I trained a demo model for 50 epochs, and I think it is overfitting.

That's expected with very few images.

You can try fine-tuning YOLOE with linear probing:

https://docs.ultralytics.com/models/yoloe/#__tabbed_1_2

**Comment by Hewgo2545** [comment]
“Can you recommend a model that achieves high accuracy with little training?” If not YOLO model. 
As i found on the internet, i found Mobilenet SSD. It will work??

**Comment by Y-T-G** [comment]
You can check YOLOE I linked above
